fix(transformer/module_runner): handle re-export as deps

hi-ogawa opened the pull request:
There wasn't a test case for this on Vite, but missing re-export deps is spotted as a test case on Vitest https://github.com/vitest-dev/vitest/pull/7646.

I'll add a test case on Vite https://github.com/vitejs/rolldown-vite/pull/85 and the same one is added here.

hi-ogawa commented:
* **#9696** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9696?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9696?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=9696).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

Dunqing reviewed (approved):
Thanks!

Dunqing commented:
### Merge activity

* **Mar 12, 3:57 AM EDT**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

hi-ogawa committed:
fix(transformer/module_runner): handle re-export as deps (#9696)

There wasn't a test case for this on Vite, but missing re-export deps is spotted as a test case on Vitest https://github.com/vitest-dev/vitest/pull/7646.

I'll add a test case on Vite https://github.com/vitejs/rolldown-vite/pull/85 and the same one is added here.
